iamctl/pkg/applications/applicationUtils.go

```diff
@@ -87,8 +100,76 @@ func getAppList() (spIdList []Application) {
 
 func getAppKeywordMapping(appName string) map[string]interface{} {
 
-	if utils.TOOL_CONFIGS.ApplicationConfigs != nil {
-		return utils.ResolveAdvancedKeywordMapping(appName, utils.TOOL_CONFIGS.ApplicationConfigs)
+	if utils.KEYWORD_CONFIGS.ApplicationConfigs != nil {
+		return utils.ResolveAdvancedKeywordMapping(appName, utils.KEYWORD_CONFIGS.ApplicationConfigs)
+	}
+	return utils.KEYWORD_CONFIGS.KeywordMappings
+}
+
+func isAuthenticationApp(fileData string) (bool, error) {
+
+	config, err := unmarshalAuthConfig([]byte(fileData))
+	if err != nil {
+		return false, err
+	}
+
+	for _, requestConfig := range config.InboundAuthenticationConfig.InboundAuthenticationRequestConfigs {
+		if strings.ToLower(requestConfig.InboundAuthType) == utils.OAUTH2 {
+			return true, nil
+		}
+	}
+	return false, nil
+}
+
+func checkInboundAuthKey(fileData []byte) (bool, error) {
+
+	config, err := unmarshalAuthConfig(fileData)
+	if err != nil {
+		return false, err
+	}
+
+	for _, requestConfig := range config.InboundAuthenticationConfig.InboundAuthenticationRequestConfigs {
+		if requestConfig.InboundAuthKey == utils.SERVER_CONFIGS.ClientId {
+			return true, nil
+		}
+	}
+	return false, nil
+}
+
+func unmarshalAuthConfig(data []byte) (AuthConfig, error) {
+	var config AuthConfig
+	err := yaml.Unmarshal(data, &config)
+	return config, err
+}
+
+func maskOAuthConsumerSecret(fileContent []byte) []byte {
+
+	// Find and replace the value of oauthConsumerSecret with asterisks
+	maskedValue := "'********'"
+	pattern := "(?m)(^\\s*oauthConsumerSecret:\\s*)null\\s*$"
+	re := regexp.MustCompile(pattern)
+	maskedContent := re.ReplaceAllString(string(fileContent), "${1}"+maskedValue)
+
+	return []byte(maskedContent)
+}
+
+func IsManagementApp(file os.FileInfo, importFilePath string) bool {
```

Comment:
Instead of returning false during error handling, let's return an error. If not, if an error occurs, we will be deleting the management app. Let's catch that error and log an appropriate message.

Reply:
Added error handling and changed the method name